### ISSUE TITLE
[Findability] User can't determine next steps in a flow.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#75980]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** govcio-vfep
**Product:** GI Bill Comparison Tool
**Feature:** Specialized Mission Filter

## Findings details
**VA.gov Experience Standard - issue:** User can't determine next steps in a flow.
**VA.gov Experience Standard - category:** Findability
**Launch-blocking:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

Screen readers may announce each of the filter type headings inconsistently, potentially assigning an incorrect heading level.

Screen reader users often have their computers announce a list of headings on the page with the heading level hierarchy as a way of scanning the page to jump to the appropriate section. This interaction pattern depends on the heading level being faithfully communicated to the user via their screen reader; otherwise, they may lose track of how different groupings of content are related to each other.

As currently coded, different screen readers are likely to announce the heading structure differently --- some correctly, some incorrectly.

<details><summary>Link, screenshot or steps to recreate</summary>

The main issue here is that the `h3` headings have an unnecessary `aria-level` attribute, eg:
```
<h3 class="school-types-label" aria-level="2">School types</h3>
```

The `h3` and the `aria-level="2"` are in conflict, and different screen readers may give priority inconsistently.

</details>

## Recommended action
At a minimum, there's no reason these headings need the `aria-level`. `<h3 class="school-types-label">School types</h3>` is sufficient.

But thinking bigger, I don't know that the headings are all that critical to navigating the filters --- or, rather, they wouldn't be if this were coded as a more typical `fieldset`/`legend` pattern. Something like:
```
<fieldset>
   <legend>School types</legend>
   ...
</fieldset>
```
... would likely be sufficient for navigating through these filters. If you're up for a bigger refactor, I think you can simplify the heading structure.

## References

- **WCAG Success Criteria:** [WCAG 2.4.6 Headings and Labels - Level AA](https://www.w3.org/TR/WCAG21/#headings-and-labels)
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [2: High. Should be fixed in 1-2 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-2)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** Platform OCTO leadership asks that we document all issues we identify. If you believe that this issue is out of scope or not your team's responsibility, it is still your responsibility to resolve the issue. Please comment on Slack in your team channel tagging `@platform-governance-team-members`, providing an explanation and who you believe is responsible. Governance team will research and follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.
